include/circuitous/Exalt/States.hpp

```diff
@@ -0,0 +1,217 @@
+/*
+ * Copyright (c) 2023 Trail of Bits, Inc.
+ */
+
+#pragma once
+
+#include <circuitous/Lifter/Context.hpp>
+#include <circuitous/IR/Intrinsics.hpp>
+
+#include <circuitous/Exalt/Common.hpp>
+
+#include <circuitous/Util/Warnings.hpp>
+#include <circuitous/Support/Log.hpp>
+#include <circuitous/Support/Check.hpp>
+
+#include <remill/Arch/Arch.h>
+#include <remill/BC/Lifter.h>
+
+#include <unordered_map>
+#include <vector>
+
+namespace circ::exalt
+{
+    // Provides wrapper around syscall related things similar to what we have for
+    // remill State.
+    struct syscall_submodule
+    {
+        // Technically we cannot return more values, therefore what we do instead is
+        // `%T = submodule( %state_in, %eax, %ebx, %ecx )`
+        // followed by a bunch of extracts from `%T` to have a nice api.
+        value_t submodule;
+        std::vector< value_t  > resutls;
+
+        static const inline std::array< std::size_t, 4 > entries =
+        {
+            8, 32, 32, 32
+        };
+
+        static inline const std::vector< std::string > syscall_regs =
```

Comment:
why not array?

Reply:
Because I was not sure how many elements we shall need (to ease prototyping). Will fix this once we support syscalls.